R/intervals.R

```diff
@@ -0,0 +1,625 @@
+#' @title Generate columns to help describe numeric variable intervals
+#'
+#' @description demUtils assumes numeric interval variables are grouped into
+#' left-closed, right-open intervals. \eqn{a <= x < b}. Each interval can be
+#' described by their endpoints from which interval lengths and nicer formatted
+#' interval names can be created.
+#'
+#' @param dt \[`data.table()`\]\cr
+#'   `col_stem`-specific data.
+#' @param id_cols \[`character()`\]\cr
+#'   ID columns that uniquely identify each row of `dt`. This must include
+#'   `{col_stem}_start`.
+#' @param col_stem \[`character(1)`\]\cr
+#'   Base name of the numeric variable column. Does not include the '_start',
+#'   '_end' etc. suffix.
```

Comment:
I think `_start` and `_end` should be turned into a parameter `stem_suffix = c("_start", "_end")`

Reply:
This is a good idea. I'll either make these changes now or start an issue.

Reply:
Made this an issue